src/detail/Application.hxx

```diff
@@ -16,6 +16,8 @@
 
 #include <string>
 
+#include "boost/asio.hpp"
```

Comment:
Why not `gethostname` or even `getenv("HOSTNAME")`? I'm not sure if including Boost is a good idea here...

Reply:
I can go for `getenv("HOSTNAME")` if we prefer. I used Boost since it's already in the dependencies. But the environmental variable should be a pretty solid solution too.